src/NuGet.Core/NuGet.Versioning/VersionRangeFactory.cs

```diff
@@ -59,6 +59,11 @@ public static VersionRange Parse(string value)
         /// </summary>
         public static VersionRange Parse(string value, bool allowFloating)
         {
+            if (value == null)
```

Comment:
Why remove it form `Parse`?

Reply:
I don't understand what you mean. I have removed it from `TryParse` and added it to `Parse` as to not regress a unit test. It is not necessary to change the behavior of `Parse` when `null` is passed. That is, `ArgumentNullException` is thrown, which is the previous behavior as well.

Reply:
yeah makes sense! I thought it other way around.